### ISSUE TITLE
Where to add expanded info for new devs?

### ISSUE THREAD

**Issue by stilez** [opnsense/docs#216]
The existing docs are extremely terse. They're good for a brief example, but leave so much untouched-on, that the learning curve is unnecessarily steep once off these basics, for less experienced or new developers, and those not previously using phalcon.

Additionally OPNsense project uses a customised file style, with xml not databases, for which even fewer references exist. Even basics such as "what types exist in a form" or "how are fields organised and collated to get from xml to form?" are hard to find.  The main source on available types is the source code file ` src/opnsense/mvc/app/views/layout_partials/form_input_tr.volt`, which isn't at all easy to locate unless one knows a bit already and what to search for, and definitely isn't readily found by a completely new-to-phalcon dev who "just wants to get going".  And there's not much at all about our xml files because thats a customisation.  

It's often easy to overlook what's needed, hard to learn, or not obvious, once it's become familiar. (That's not in any way meant as a crtitique, it's a recognition of something almost inevitable and almost universal for new devs on many projects).  As I'm a new dev, and seem (surprisingly!) to be coping with learning this, and starting to push a few PRs, I'd like to start adding some more targeted brief documentation and examples to ease the way for other new to these things, to simplify the learning curve. 

It would be a kind of "New Developers Guide" (or section) taking up where "hello world" stops, to fill the gap between "hello world" and a realistic extension or core code PR, overviewing practical creating and modifying legacy and new-style code, extensions, etc - which I suspect will be useful. 

**Tl;Dr - If that's okay, can you let me know where to put them in the dev guide, and a preferred title for now?**
